fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -6331,25 +6332,31 @@ public void setHasForbitGlobalDict(String dbName, String tableName, boolean isFo
         Map<String, String> property = new HashMap<>();
         Database db = getDb(dbName);
         if (db == null) {
-            throw new DdlException("the DB " + dbName + "isn't  exist");
-        }
-
-        Table table = db.getTable(tableName);
-        if (table == null) {
-            throw new DdlException("the DB " + dbName + " table: " + tableName + "isn't  exist");
+            throw new DdlException("the DB " + dbName + " is not exist");
         }
+        db.readLock();
+        try {
+            Table table = db.getTable(tableName);
+            if (table == null) {
+                throw new DdlException("the DB " + dbName + " table: " + tableName + "isn't  exist");
+            }
 
-        if (table instanceof OlapTable) {
-            OlapTable olapTable = (OlapTable) table;
-            olapTable.setHasForbitGlobalDict(isForbit);
-            if (isForbit) {
-                property.put(PropertyAnalyzer.ENABLE_LOW_CARD_DICT_TYPE, PropertyAnalyzer.DISABLE_LOW_CARD_DICT);
-            } else {
-                property.put(PropertyAnalyzer.ENABLE_LOW_CARD_DICT_TYPE, PropertyAnalyzer.ABLE_LOW_CARD_DICT);
+            if (table instanceof OlapTable) {
+                OlapTable olapTable = (OlapTable) table;
+                olapTable.setHasForbitGlobalDict(isForbit);
+                if (isForbit) {
+                    property.put(PropertyAnalyzer.ENABLE_LOW_CARD_DICT_TYPE, PropertyAnalyzer.DISABLE_LOW_CARD_DICT);
+                    IDictManager.getInstance().disableGlobalDict(olapTable.getId());
```

Comment:
I think we needn't  `disableGlobalDict` and  `enableGlobalDict` methods in IDictManager.  only keep tableProperty in  OlapTable is enough, simple and high performance.

Reply:
if we want use tableProperty in OlapTable. we need acquire the lock when we need update the properties.